server/src/test/java/io/crate/integrationtests/AlterTableIntegrationTest.java

```diff
@@ -269,4 +270,29 @@ private void execute_statements_that_drop_and_add_column_with_the_same_name_agai
             String.format(Locale.ENGLISH, rawRows, newColumnOid, newColumnOid + 1, newColumnOid + 2)
         );
     }
+
+    @Test
+    public void test_alter_table_can_add_column_after_dropping_column_with_max_known_position() {
+        execute("create table t(a int, o object AS(oo object AS(a int)))");
+
+        // Initially position 4 is taken by column o['oo']['a']
+        execute("select column_name, ordinal_position from information_schema.columns where table_name = 't' AND ordinal_position = 4");
+        assertThat(printedTable(response.rows())).isEqualTo(
+            """
+            o['oo']['a']| 4
+            """);
+
+        // Drop column having sub-columns, o['oo']['a'] is dropped implicitly.
+        execute("alter table t drop column o['oo']");
+
+        //o['oo']['a'] has max position 4 but it's gone from the mapping.
+        // We take to account dropped column positions, but only o['oo'] information is stored in the metadata.
+        // Hence, next added column gets position of the o['oo']['a'].
+        execute("alter table t add column b int");
```

Comment:
Without the fix `ADD COLUMN` fails with duplicate position assignment error.

Reply:
Can you please create a simple test case for top-level columns as well/instead?
It's a bit hard to understand why this test would fail without the fix. Also this scenario is not completely related to the fix, the fix is responsible to keep the max position on 3 (including the dropped col) but not related to the missing sub-columns.

Reply:
I replaced this test by a simpler one.